kinoml/core/proteins.py

```diff
@@ -137,18 +139,21 @@ def from_hgnc(cls, identifier, **kwargs):
     def from_name(cls, identifier, **kwargs):
         import requests
         import tempfile
-        from MDAnalysis import Universe
+        import MDAnalysis as mda
         from pathlib import Path
+        from appdirs import user_cache_dir
+
+        p = f"{user_cache_dir()}/{identifier}.pdb"
 
         url = f"https://files.rcsb.org/download/{identifier}.pdb"
         response = requests.get(url)
 
-        with tempfile.NamedTemporaryFile(suffix=".pdb") as temp_file:
-            temp_file.write(response.content)
-            u = Universe(temp_file.name)
-            p = Path(temp_file.name)
+        with open(p, "wb") as pdb_file:  # saving the pdb to cache
+            pdb_file.write(response.content)
+
+        u = mda.fetch_mmtf(identifier)
```

Comment:
Are you fetching the PDB twice? Why do we need that?

Reply:
This adds the query PDB structure the path. I guess we could load the `Universe` from that path rather than fetching again?